File: week06/explore_phila_data/extract_phl_opa_properties.py
import pathlib
import requests

DATA_DIR = pathlib.Path(__file__).parent / 'raw_data'

# Download the OPA Properties data as a CSV
url = 'https://opendata-downloads.s3.amazonaws.com/opa_properties_public.csv'
filename = DATA_DIR / 'phl_opa_properties.csv'

response = requests.get(url)
response.raise_for_status()

with open(filename, 'wb') as f:
    f.write(response.content)

print(f'Downloaded {filename}')

File: week06/explore_phila_data/extract_phl_opa_properties.mjs
import fetch from 'node-fetch';
import fs from 'fs/promises';

const url = 'https://opendata-downloads.s3.amazonaws.com/opa_properties_public.csv';
const filename = 'phl_opa_properties.csv';

const response = await fetch(url);
if (!response.ok) {
  throw new Error(`HTTP error! status: ${response.status}`);
}

await fs.writeFile(filename, await response.text());

console.log(`Downloaded ${filename}`);
